generate/generate_hcl_test.go

```diff
@@ -246,6 +246,177 @@ func TestGenerateHCL(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "generate HCL for all stacks importing common",
+			layout: []string{
+				"s:stacks/stack-1",
+				"s:stacks/stack-2",
+			},
+			configs: []hclconfig{
+				{
+					path: "/common",
+					add: hcldoc(
+						generateHCL(
+							labels("backend.tf"),
+							content(
+								backend(
+									labels("test"),
+									expr("prefix", "global.backend_prefix"),
+								),
+							),
+						),
+						generateHCL(
+							labels("locals.tf"),
+							content(
+								locals(
+									expr("stackpath", "terramate.path"),
+									expr("local_a", "global.local_a"),
+									expr("local_b", "global.local_b"),
+									expr("local_c", "global.local_c"),
+									expr("local_d", "tm_try(global.local_d.field, null)"),
+								),
+							),
+						),
+						generateHCL(
+							labels("provider.tf"),
+							content(
+								provider(
+									labels("name"),
+									expr("data", "global.provider_data"),
+								),
+								terraform(
+									requiredProviders(
+										expr("name", `{
+										source  = "integrations/name"
+										version = global.provider_version
+									}`),
+									),
+								),
+								terraform(
+									expr("required_version", "global.terraform_version"),
+								),
+							),
+						),
+					),
+				},
+				{
+					path: "/stacks/stack-1",
+					add: hcldoc(
+						importy(
```

Comment:
Im literally LoLing here :laughing: (pretty sure JS will add this at some point) :rofl: